File: src/main/java/com/baulsupp/oksocial/util/ProtocolUtil.java
package com.baulsupp.oksocial.util;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import okhttp3.Protocol;

public class ProtocolUtil {
  public static List<Protocol> parseProtocolList(String protocols) {
    List<Protocol> protocolValues = new ArrayList<>();

    try {
      for (String protocol : protocols.split(",")) {
        protocolValues.add(Protocol.get(protocol));
      }
    } catch (IOException e) {
      throw new IllegalArgumentException(e);
    }

    if (!protocolValues.contains(Protocol.HTTP_1_1)) {
      protocolValues.add(Protocol.HTTP_1_1);
    }

    return protocolValues;
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/util/ProtocolUtil.kt
package com.baulsupp.oksocial.util

import okhttp3.Protocol
import java.io.IOException
import java.util.*

object ProtocolUtil {
    fun parseProtocolList(protocols: String): List<Protocol> {
        val protocolValues = ArrayList<Protocol>()

        try {
            protocols.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray().mapTo(protocolValues) { Protocol.get(it) }
        } catch (e: IOException) {
            throw IllegalArgumentException(e)
        }

        if (!protocolValues.contains(Protocol.HTTP_1_1)) {
            protocolValues.add(Protocol.HTTP_1_1)
        }

        return protocolValues
    }
}
